craft_parts/utils/path_utils.py

```diff
@@ -80,8 +87,33 @@ def get_partition_and_path(path: FlexiblePath) -> PartitionPathPair:
     str_path = str(path)
 
     if _has_partition(str_path):
-        if "/" not in str_path:
-            return PartitionPathPair(str_path.strip("()"), cast(FlexiblePath, ""))
-        partition, inner_path = str_path.split("/", maxsplit=1)
+        partition, inner_path = _split_partition_and_inner_path(str_path)
         return PartitionPathPair(partition.strip("()"), path.__class__(inner_path))
+
     return PartitionPathPair("default", path)
+
+
+def _split_partition_and_inner_path(str_path: str) -> Tuple[str, str]:
+    """Split a path with a partition into a partition and inner path.
+
+    :param str_path: A string representation of a filepath to split.
+
+    :returns: A tuple containing the partition and inner path. If there is no inner
+    path, the second element will be an empty string.
+    """
+    # split regular partitions
+    if re.match(HAS_PARTITION_REGEX, str_path):
+        if "/" in str_path:
+            # split into partition and inner_path
+            partition, inner_path = str_path.split("/", maxsplit=1)
+            # remove extra forward slashes between the partition and inner path
+            return partition, inner_path.lstrip("/")
```

Comment:
This was a bug I uncovered, fixed, and tested as part of this PR

The bug is that `get_partitioned_path("(default)//foo")` returned `/foo` instead of `default/foo`

Reply:
Oh that's sneaky. Well spotted!